### PR TITLE
Functions home task part - 1. Refactored collections home task.

### PR DESCRIPTION
**Pull request by DenisKrupkin**
Home task for functions part 1.

### CODE REVIEW

File: Functions/functions_collections_home_task_edited.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# Function for generation of single dictionary with defined ranges of random keys and values
+def generate_dictionary_with_random_keys_and_values(
+        set_of_keys,
+        start_of_value, end_of_value,
+        start_of_total_count, end_of_total_count,
+):
+    final_dictionary = {random.choice(set_of_keys): random.randint(start_of_value, end_of_value)
+                        for _ in range(random.randint(start_of_total_count, end_of_total_count))}
+    return final_dictionary
```
Comment (arastopchin) [inline Functions/functions_collections_home_task_edited.py]:
```suggestion
def generate_random_keys_values_dict(keys, min_value, max_value, min_count, max_count):
    """
    Function generates a dictionary with defined ranges of random keys and values.
    @param keys: possible keys set
    @param min_value: min possible number for value
    @param max_value: max possible number for value
    @param min_count: min count of values to generate
    @param max_count: max count of values to generate
    @return: random key to random value dict with random count of {key: value} pairs
    """
    return {random.choice(keys): random.randint(min_value, max_value) for _ in range(random.randint(min_count, max_count))}
```
It's just an example how funciton may look like. Here I added a docstring (pls read what it's about), renamed function itself, renamed arguments. Try to not use 'of', 'with' and another prepositions. I hope I showed you a good example of naming. Also never use such things as 'final', 'pre_final' bla bla it makes no sense. Try to use such variable names which will makes sense for reader. If your function does only one thing you can to not name variable at all, just return it in place.

File: Functions/functions_collections_home_task_edited.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+# Function for generation of list with defined count of random generated dictionaries
+def create_list_with_random_dictionaries(start_of_count_of_dicts, end_of_count_of_dicts):
```
Comment (arastopchin) [inline Functions/functions_collections_home_task_edited.py]:
```suggestion
def create_random_dictionaries_list(min_dicts_count, max_dicts_count, keys, min_value, max_value, min_values_count, max_values_count):
"""
some docstring
"""
```
You pass (string.ascii_lowercase, 0, 100, 1, 100) but it's not mentioned in name of function, may be you should also pass these values as arguments? In this case if you want to use this function for generatin random dictionaries with another keys and values you can do so.

**Review by arastopchin — CHANGES_REQUESTED** [review]
Please follow recommendations.

File: Functions/functions_collections_home_task_edited.py (hunk 1 above)
Comment (DenisKrupkin) [reply Functions/functions_collections_home_task_edited.py]: Tried to make all functions using this code stile.

File: Functions/functions_collections_home_task_edited.py (hunk 2 above)
Comment (DenisKrupkin) [reply Functions/functions_collections_home_task_edited.py]: Fixed.

File: Functions/functions_strings_home_task_edited.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    @param target_text: text to normalize
+    @param paragraph_splitter: paragraph splitter for target text
+    @return: text with corrected letters case and paragraphs
+    """
+    # Split text by sentences and without whitespaces at the beginning and the capitalizes it:
+    target_text = '. '.join(item.lstrip().capitalize() for item in target_text.split('. '))
+    target_text_list = []  # Create list for store normalized sentences
+    for sentence in target_text.split('.\n'):  # Split text by paragraphs
+        sentence = sentence.strip()  # Delete whitespaces
+        if sentence[0].islower() and sentence[0].isalpha():  # Check is sentence starts with lower case letter
```
Comment (arastopchin) [inline Functions/functions_strings_home_task_edited.py]: Why do you do this check? If remove it nothing will change.

File: Functions/functions_strings_home_task_edited.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+ 
+
+  it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE.
+
+ 
+
+  last iz TO calculate nuMber OF Whitespace characteRS in this Tex. caREFULL, not only Spaces, but ALL whitespaces. I got 87."""
+
+# Generate normalized text from variable.
+normalized_text = normalize_paragraphs_letters_case(corrupted_text, '\n\n\n\n')
```
Comment (arastopchin) [inline Functions/functions_strings_home_task_edited.py]: I would replace \n\n\n\n with some variable to not repeat it. 
Global variables usually use UPPERCASE in Python.

**Review by arastopchin — APPROVED** [review]
Quite good! If you want you can correct my notes, but it's not really needed.

File: Functions/functions_strings_home_task_edited.py (hunk 3 above)
Comment (DenisKrupkin) [reply Functions/functions_strings_home_task_edited.py]: Just tried to make logic a little bit smarter) but actually I agree with you)

File: Functions/functions_strings_home_task_edited.py (hunk 4 above)
Comment (DenisKrupkin) [reply Functions/functions_strings_home_task_edited.py]: Added global variable.

**Pull request merged**